### PR TITLE
Simplify logic for convertSnapshotToITree api for taking summary during attach container of detached container

### LINKED ISSUE #4278: Simplify logic for convertSnapshotToITree api
Refer this: https://github.com/microsoft/FluidFramework/blob/ee14a3966db4a44053e760b3b8c981d0595a66b4/packages/runtime/datastore/src/utils.ts#L10
`blobMapInitial` simply contains entries one-to-one with `snapshot.blobs`, so basically converted from plain JS object to Map.

Then loop through those entries in `blobMapInitial` and add them to `blobMapFinal` if they exist in `blobMapInitial` (redundant?). Seems like `blobMapInitial` and `blobMapFinal` are the same.

_Originally posted by @arinwt in https://github.com/microsoft/FluidFramework/pull/4230#discussion_r519020998_

### PR DESCRIPTION
**Pull request by jatgarg**
Fixes: https://github.com/microsoft/FluidFramework/issues/4278
Remove redundant api as both are now same.
The entries in blobs are supposed to be blobPath -> blobId and blobId -> blobValue and we want to push blobPath to blobValue in tree entries. These blobIds are unique which are generated when taking snapshot in detached container using serialize api.
There would never be a case where a -> b, b -> c and c->d.

### CODE REVIEW

File: packages/runtime/runtime-utils/src/summaryUtils.ts [hunk 1]
```diff
@@ -214,27 +214,25 @@ export function convertToSummaryTree(
...
 
-    const blobMapInitial = new Map(Object.entries(snapshot.blobs));
-    const blobMapFinal = new Map<string, string>();
-    for (const [key, value] of blobMapInitial.entries()) {
-        if (blobMapInitial.has(value)) {
-            blobMapFinal[key] = blobMapInitial.get(value);
+    const builder = new SummaryTreeBuilder();
+    for (const [key, value] of Object.entries(snapshot.blobs)) {
+        // The entries in blobs are supposed to be blobPath -> blobId and blobId -> blobValue
+        // and we want to push blobPath to blobValue in tree entries.
```
Comment (arinwt) [inline packages/runtime/runtime-utils/src/summaryUtils.ts:229]: Great comment!

**Review by arinwt — APPROVED** [review]
PR is good. I am curious why they are stored this way? path -> id and id -> value in the same object?? but that's a different question.

**Comment by jatgarg** [comment]
> PR is good. I am curious why they are stored this way? path -> id and id -> value in the same object?? but that's a different question.

This is the snapshot taken for drafts by serialize api in detach container. When we normally load from a snapshot we have blobpath to blobl id mappings and then the container request blob corresponding to that id from storage to have the blob value. But in drafts when we load detached container from snapshot nothing exists on storage, so we stored similar mappings inside blobs(which in attached case are on storage) so that we don't have to do anything special in normal datastore/dds load path. In this api we use the same snapshot to get summary when attaching the same container in case a particular datastore/dds is not loaded, so in actual snapshot we eliminate the entries and just have blobpath to blobvalue just like in normal snapshot.

**Comment by arinwt** [comment]
@jatgarg does this mean that this conversion function is specialized for detached summary? If so, does that mean that @agarwal-navin shouldn't be using it to convert blobs of regular summaries?

**Comment by jatgarg** [comment]
@arinwt  Yes, it is specialized and I have added that in comment too. As I checked I don't think @agarwal-navin is using it.

**Pull request merged**